internal/service/encryptionatrest/data_source_schema.go

```diff
@@ -0,0 +1,184 @@
+						MarkdownDescription: "Resource path that displays the key version resource ID for your Google Cloud KMS.",
+					},
+					"service_account_key": schema.StringAttribute{
+						Computed:            true,
+						Sensitive:           true,
+						Description:         "JavaScript Object Notation (JSON) object that contains the Google Cloud Key Management Service (KMS). Format the JSON as a string and not as an object.",
+						MarkdownDescription: "JavaScript Object Notation (JSON) object that contains the Google Cloud Key Management Service (KMS). Format the JSON as a string and not as an object.",
+					},
+					"valid": schema.BoolAttribute{
+						Computed:            true,
+						Description:         "Flag that indicates whether the Google Cloud Key Management Service (KMS) encryption key can encrypt and decrypt data.",
+						MarkdownDescription: "Flag that indicates whether the Google Cloud Key Management Service (KMS) encryption key can encrypt and decrypt data.",
+					},
+				},
+				Computed:            true,
+				Description:         "Details that define the configuration of Encryption at Rest using Google Cloud Key Management Service (KMS).",
+				MarkdownDescription: "Details that define the configuration of Encryption at Rest using Google Cloud Key Management Service (KMS).",
+			},
+			"project_id": schema.StringAttribute{
+				Required:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+			},
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+type TFEncryptionAtRestDSModel struct {
+	AzureKeyVaultConfig  *TFAzureKeyVaultConfigModel `tfsdk:"azure_key_vault_config"`
+	AwsKmsConfig         *TFAwsKmsConfigModel        `tfsdk:"aws_kms_config"`
+	GoogleCloudKmsConfig *TFGcpKmsConfigModel        `tfsdk:"google_cloud_kms_config"`
+	ID                   types.String                `tfsdk:"id"`
+	ProjectID            types.String                `tfsdk:"project_id"`
+}
+
+func NewTFEncryptionAtRestDSModel(projectID string, encryptionResp *admin.EncryptionAtRest) *TFEncryptionAtRestDSModel {
```

Comment:
This model is identical to TFEncryptionAtRestRSModel right? We could use the same one and avoid defining this new conversion method as well. In this case we could rename to `TFEncryptionAtRestModel`

Reply:
They're not identical. The difference is that TFEncryptionAtRestRSModel has array objects for nested AWS, Azure, GCP configs. That's because the resource was migrated from TF SDK to TPF so we had to keep some behavior to avoid non-empty plans after that migration.

Reply:
Ohh I see, thanks for clarifying. Have my doubts if we prefer to have the schemas aligned, but understand the current data source aligns better with the API.

Reply:
Yeah I'd prefer to keep it as is, not only due to closer alignment with the API but also because having those objects as lists in the resource was something we had to do in the past because of TF SDK limitations but now that we have TPF I'd want to stay on the correct implementation which is having these as singular objects.